+nosnoc/+discrete_time_problem/Heaviside.m

```diff
@@ -939,12 +939,154 @@ function create_solver(obj, solver_options, plugin)
+                    lambda_n_values = zeros(dims.n_alpha,1);
+                    lambda_n_values(I_0_ii) = 1;
+                    lambda_p_values = zeros(dims.n_alpha,1);
+                    lambda_p_values(I_1_ii) = 1;
+                    for jj=jj:opts.N_stages
+                        for kk=(kk+1):opts.N_finite_elements(jj)
+                            for ll=1:opts.n_s % TODO(@anton) handle GL
+                                obj.w.alpha(jj,kk,ll).init = alpha_values;
+                                obj.w.lambda_n(jj,kk,ll).init = lambda_n_values;
+                                obj.w.lambda_p(jj,kk,ll).init = lambda_p_values;
+                            end
+                            t_curr = t_curr + opts.h_k(jj);
+                            if t_curr > t_end
+                                done_stage = true;
+                                break
+                            end
+                        end
+                        if done_stage
+                            break
+                        end
+                        kk = 0; % Reset fe counter
+                    end
+                    % handle entering regions TODO(@anton) verify this is correct
+                    if ii ~= active_set.get_n_steps()
+                        I_0_next = active_set.I_0{ii+1};
+                        I_1_next = active_set.I_1{ii+1};
+                        I_0_leaving = setdiff(I_0_next, I_0_ii);
+                        I_1_leaving = setdiff(I_1_next, I_1_ii);
+                        lambda_n_values(I_0_leaving) = 0;
+                        lambda_p_values(I_1_leaving) = 0;
+                        obj.w.lambda_n(jj,kk,ll).init = lambda_n_values;
+                        obj.w.lambda_p(jj,kk,ll).init = lambda_p_values;
+                    end
+                end
+            else
+                jj = 1; % Control stage index
+                kk = 0; % FE index
+                t_curr = 0;
+                % Go through the active set by times
+                for ii=1:active_set.get_n_steps()
+                    done_stage = false;
+                    stage_end = active_set.stages{ii};
+                    
+                    N_end = stage_end(1);
+                    n_end = stage_end(2);
+
+                    I_0_ii = active_set.I_0{ii};
+                    I_1_ii = active_set.I_1{ii};
+                    I_free_ii = active_set.I_free{ii};
+
+                    alpha_values = zeros(dims.n_alpha,1);
+                    alpha_values(I_0_ii) = 0;
+                    alpha_values(I_1_ii) = 1;
+                    alpha_values(I_free_ii) = 0.5; % TODO(@anton) is there a generically better formula?
+                    lambda_n_values = zeros(dims.n_alpha,1);
+                    lambda_n_values(I_0_ii) = 1;
+                    lambda_p_values = zeros(dims.n_alpha,1);
+                    lambda_p_values(I_1_ii) = 1;
+
+                    for jj=jj:N_end
```

Comment:
few lines of comments in more complicated chucks to explain what will be done would be very welcome.